libs/data-mapper/src/lib/mapDefinitions/__test__/MapDefinitionSerializer.spec.ts

```diff
@@ -1,1888 +1,3705 @@
-        (child) => child.name === 'LoopingWithIndex'
-      ) as SchemaNodeExtended;
-      const directAccessId = createReactFlowFunctionKey(directAccessPseudoFunction);
-      const indexFunctionId = createReactFlowFunctionKey(indexPseudoFunction);
-      const ifId = createReactFlowFunctionKey(ifPseudoFunction);
-      const greaterThanId = createReactFlowFunctionKey(greaterThanFunction);
-      const mapDefinition: MapDefinitionEntry = {};
-      const connections: ConnectionDictionary = {};
-
-      // Just confirm the mock hasn't changed
-      expect(sourceNode).toBeDefined();
-      expect(targetNode).toBeDefined();
-
-      const parentSourceNode = sourceNode.children[0];
-      const parentTargetNode = targetNode.children[0].children[2];
-
-      // Parent source to index
-      setConnectionInputValue(connections, {
-        targetNode: indexPseudoFunction,
-        targetNodeReactFlowKey: indexFunctionId,
-        findInputSlot: true,
-        value: {
-          reactFlowKey: addReactFlowPrefix(parentSourceNode.key, SchemaType.Source),
-          node: parentSourceNode,
-        },
-      });
-
-      // Connect index to parent target
-      setConnectionInputValue(connections, {
-        targetNode: parentTargetNode,
-        targetNodeReactFlowKey: addReactFlowPrefix(parentTargetNode.key, SchemaType.Target),
-        findInputSlot: true,
-        value: {
-          reactFlowKey: indexFunctionId,
-          node: indexPseudoFunction,
-        },
-      });
-
-      // Connect to direct access
-      setConnectionInputValue(connections, {
-        targetNode: directAccessPseudoFunction,
-        targetNodeReactFlowKey: directAccessId,
-        findInputSlot: true,
-        value: {
-          reactFlowKey: indexFunctionId,
-          node: indexPseudoFunction,
-        },
-      });
-      setConnectionInputValue(connections, {
-        targetNode: directAccessPseudoFunction,
-        targetNodeReactFlowKey: directAccessId,
-        findInputSlot: true,
-        value: {
-          reactFlowKey: addReactFlowPrefix(parentSourceNode.key, SchemaType.Source),
-          node: parentSourceNode,
-        },
-      });
-      setConnectionInputValue(connections, {
-        targetNode: directAccessPseudoFunction,
-        targetNodeReactFlowKey: directAccessId,
-        findInputSlot: true,
-        value: {
-          reactFlowKey: addReactFlowPrefix(parentSourceNode.children[0].key, SchemaType.Source),
-          node: parentSourceNode.children[0],
-        },
-      });
-
-      // Direct access to target
-      setConnectionInputValue(connections, {
-        targetNode: parentTargetNode.children[1],
-        targetNodeReactFlowKey: addReactFlowPrefix(parentTargetNode.children[1].key, SchemaType.Target),
-        findInputSlot: true,
-        value: {
-          reactFlowKey: directAccessId,
-          node: directAccessPseudoFunction,
-        },
-      });
+  describe('JSON to JSON', () => {
```

Comment:
The first 4 tests (the ones not skipped) are actually new and updated for JSON. All the skipped ones are copied from XML and ignored for now